### PR TITLE
This was discussed at our last meeting and is being PR'd for discussion.

### PR DESCRIPTION
**Pull request by FlipperPA**
This potentially addresses https://github.com/code4recovery/spec/issues/17.

### CODE REVIEW

**Comment by FlipperPA** [comment]
Just to get discussion started on specifics, I don't expect this to be the final solution.

For a first question:

Do we separate into three option?

- Active Meeting? `Active` / `Temp Closed` / `Perm Closed`
- In Person? `True` / `False`
- Online? `True` / `False`

This might be better, as it avoids overloading fields, and any time I've done that in the past... I've regretted it.

**Comment by tintinnabulate** [comment]
I think it's a good idea, provided the Meeting Validator checks for constraints (for example):

``` ada
if not (In_Person and then Online) then
   pragma Assert (Meeting_Status /= Active)
```

Also, have you considered implementing a JSON schema? https://json-schema.org/

**Comment by tech2serve** [comment]
I think I'm still leaning towards splitting the data, and I believe two fields are sufficient (but see my comments below about whether this is really actually a good idea):

1. Attendance Option (as a string): 
    a. In-person
    b. Hybrid
    c. Online
2. Status (as a string)
    a. Active (default)
    b. Temporarily suspended
    c. Archived

My logic for splitting the data into multiple fields is that makes it easier to the user to manage the database (even if implementation is a bit more complex). For example, the webmaster suspends a meeting but doesn't have to change the original attendance option. This way, say three months later, they are not having to try to remember (nor did they have to record the original format in notes or somewhere else) if the meeting was online only or a hybrid.

As far as boolean vs descriptive fields, I've personally had more luck implementing the latter across multiple programming languages. For some reason, boolean seems trickier and ultimately less clear. I'm not saying it should be that way, just that it seems to be my experience.

_Important!_ I do believe the question of how detailed this needs to be in a data exchange specification remains valid. While I tend to favor explicit definitions, I realize they become potentially harder to change as platforms evolve. And, since Meeting Guide (MG) is our ultimate target, does the status of the meeting really need to be sent from a website/app to MG if it is suspended? Instead, the feed simply needs to NOT send the meeting if it isn't active. So, ultimately, I'm not convinced the data exchange specification needs to include status, rather just the Attendance Option. How a website (or TSML plugin) handles whether a meeting is active, suspended or gone completely (archived?) is really up to that app/website.

To summarize, I'm now leaning towards the idea of including the descriptors necessary for characterizing how the meeting is being delivered to users in the spec, but skipping the status (because only active meetings should be exchanged with MG). I also acknowledge that the use of booleans as proposed (overloading risk?) by @FlipperPA may be better for the actual data exchange. I'll defer to more experienced developers here. 

I realize I'm changing my tune a bit from Saturday's discussion, and I think this is where writing it out is helpful for me.

**Comment by brianw-area24** [comment]
I need to think about this a little more, but I'm also leaning toward 2 fields. If we do In-person, and Online, as two boolean fields, what does it mean if both fields are false? This is an invalid state, so we'd have to do something to guard against this from happening. So to me it makes sense to combine these fields into 1.

As for the Meeting status field, in WordPress, I'm not sure we need archived/temp closed as an option. Rather, wouldn't it just be a unpublished post? I'm having a hard time thinking of a time when we'd want to display a permanently closed meeting on a public web site.

The other thing I'm trying to wrap my head around, is how will these fields be handled in the search mechanism in the ui? This line with the search options is already pretty full, so how are we going to add 2 more selections?

**Comment by jcookson-infoblox** [comment]
I love this .... so this addresses the issue my area has, that of requesting a type for hybrid.  So, by doing this - it truly addresses it, because their request is not technically a "type" - but an attendance option.

**Comment by tech2serve** [comment]
> I love this .... so this addresses the issue my area has, that of requesting a type for hybrid. So, by doing this - it truly addresses it, because their request is not technically a "type" - but an attendance option.

Correct. This is the approach we're taking with TSML (Thanks @brianw-area24!) but we'll keep them also as types for backwards compatibility with MG. We're working with them to eventually change the spec.

**Comment by joshreisner** [comment]
I want to caution against changing the spec in this way, because it introduces indicators that could be in conflict. For example:

* what do you do if a meeting is `attendance_option` = `online` and yet `conference_url` and `conference_phone` are empty? is it still "online?"
* also what if a meeting is `in-person` and yet has no `address`
* I'm assuming the `TC` type goes away, but if it didn't then that's another indicator that could be in conflict

Remember, this is not what the user sees. It's perfectly fine to have your options above in the user interface, but keep a spec the way it is. The TSML-UI [demo website](https://react.meetingguide.org) has those options now without requiring a spec change. The Meeting Guide app could also adopt these metatypes.

**Comment by jcookson-infoblox** [comment]
So - tester brain went off .... In order to figure out potential conflicts, would need to know the fields in question and states ... also, something to think about.  A floating meeting is a meeting that is in-person and has no address, yet it is still a meeting.  This would violate the constraints.

**Comment by joshreisner** [comment]
in the last month there has been a lot of terrific development on both TSML and TSML UI that has demonstrated that we can achieve these objectives without introducing conflicting signals into the spec. going to close this PR but i'm happy to continue the discussion
